File: LeetCode/Easy/0700-search-in-a-binary-search-tree/0700-search-in-a-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode searchBST(TreeNode node, int val) {
        if(node.val == val) return node;

        if(node.left != null) {
            TreeNode checked = searchBST(node.left, val);
            if(checked != null) return checked;
        }

        if(node.right != null) return searchBST(node.right, val);

        return null;
    }
}

File: LeetCode/Easy/0700-search-in-a-binary-search-tree/0700-search-in-a-binary-search-tree.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun searchBST(node: TreeNode?, `val`: Int): TreeNode? {
        if(node == null) return node
        if(node.`val` == `val`) return node;

        if(node.left != null) {
            val checked = searchBST(node.left, `val`)
            if(checked != null) return checked
        }

        if(node.right != null) return searchBST(node.right, `val`)

        return null
        
    }

}